### PR TITLE
Features/redux form api adapter

### PR DESCRIPTION
**Pull request by julianrogmans**
@randycoulman ready for review

### CODE REVIEW

File: client/utils/__tests__/api-spec.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  it('converts the SUCCESS action to a type descriptor', () => {
+    const options = withDefaults({
+      types: ['R', 'S', 'F']
+    })
+
+    expect(options.types[1].type).eql('S')
+  })
+
+  it('keeps SUCCESS and FAILURE actions as type descriptor', () => {
```
Comment (randycoulman) [inline client/utils/__tests__/api-spec.js]: Should be `descriptors` (plural).  Probably my mistake where you copied this from

File: client/utils/__tests__/api-spec.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    const payload = {
+      foo_bar: 42 // eslint-disable-line camelcase
+    }
+    const response = JSON.stringify(payload)
+    const actual = options.types[1].payload(null, null, response)
+    const expected = { fooBar: 42 }
+
+    expect(actual).to.eventually.eql(expected)
+  })
+})
```
Comment (randycoulman) [inline client/utils/__tests__/api-spec.js]: We need a spec for the FAILURE payload as well.  Not your fault - your source material didn't have one either.  But we should add one here.

File: client/utils/__tests__/form-spec.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import { expect } from 'chai'
+import td from 'testdouble'
+
+import { formApiAdapter } from '../form'
+
+describe('formApiAdapter', () => {
+  const formValues = {}
+  const action = {}
```
Comment (randycoulman) [inline client/utils/__tests__/form-spec.js]: We should have something better than empty objects for these.  Again, my bad on the original.

**Comment by randycoulman** [comment]
Looks like you got it all and weeded through it correctly, but the original was missing some specs that we should add here.

File: client/utils/__tests__/form-spec.js (hunk 3 above)
Comment (julianrogmans) [inline client/utils/__tests__/form-spec.js]: I might have you help me out on this one

File: client/utils/__tests__/api-spec.js (hunk 2 above)
Comment (julianrogmans) [inline client/utils/__tests__/api-spec.js]: I have added a spec but I did it in a copy-paste way so it might need some modifications

**Comment by randycoulman** [comment]
The project code that these adapters were based on changed heavily since this PR was created.  Rather than trying to adapt this PR to the new code, we created a new PR, #42, with the new code.

**Pull request closed without merging**